sw/device/tests/otbn/otbn_rsa_test.c

```diff
@@ -0,0 +1,933 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * End-to-end RSA encryption and decryption test using OTBN.
+ *
+ * IMPORTANT: This test is not a secure, complete, or reusable implementation of
+ * RSA; it is not even close to being production-ready. It is only meant as an
+ * end-to-end test for OTBN during the bringup phase.
+ *
+ * This test loads the RSA application into OTBN, sets all required input
+ * arguments, and performs the encryption and decryption operations.
+ *
+ * To keep the test execution time reasonable some parts of the test can be
+ * disabled with the `kTestDecrypt` and `kTestRsaGreater1k` constants.
```

Comment:
You mention keeping "reasonable" text execution times, and that decryption is "significantly slower" than encryption. I know that performance is going to depend on the host system you are running the tests on, but do you have order-of-magnitude number for test execution times with different combinations of `kTestDecrypt` and `kTestRsaGreater1k`? I'm concerned that down the road, someone will have trouble distinguishing between these tests being slow and these tests hanging.

Reply:
Yeah, "reasonably" is a rather fluffy term, and we'll need to look carefully at which tests to enable where.

For simulation timings, please have a look at the execution log https://github.com/lowRISC/opentitan/pull/4071#issuecomment-722255770. The whole test with all options enabled took around 12 hours on my machine in Verilator simulation. The log also contains the cycles for the individual tests and the execution speed in simulation. 

I can see us enabling only the smallest test in Verilator simulations, and run all other tests only on FPGAs, but that's something we need to adjust when we add the configurations to CI and DV regression runs.

Reply:
I think something like the following would be helpful:

Using Verilator, running these tests with `kTestDecrypt` and `kTestRsaGreater1k` takes a couple minutes. Enabling one of `kTestDecrypt` or `kTestRsaGreater1k` adds about ten minutes to the runtime. Enabling both extends the runtime to 10-20 hours.

Reply:
I added this information to the documentation of `kTestDecrypt`.